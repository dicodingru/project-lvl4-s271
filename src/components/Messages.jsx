import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import styled from 'styled-components';
import Message from './Message';
import NewMessageForm from './NewMessageForm';

const Div = styled.div`
  width: 70%;
  flex-direction: column;
  justify-content: flex-end;
`;

const Messages = (props) => {
  const { messages } = props;
  return (
    <Div>
      {messages.map((item) => <Message key={_.uniqueId()} text={item} />)}
      <NewMessageForm />
    </Div>
  );
};

Messages.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Messages;
