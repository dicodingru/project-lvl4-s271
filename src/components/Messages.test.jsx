import React from 'react';
import renderer from 'react-test-renderer';
import Messages from './Messages';

describe('Messages component', () => {
  let component;
  const messages = ['msg1', 'msg2'];

  beforeEach(() => {
    component = shallow(<Messages messages={messages} />);
  });

  test('should render successfully', () => {
    expect(component.exists()).toEqual(true);
  });

  test('should render list of two messages', () => {
    expect(component.find('Message').length).toBe(2);
  });

  test('should render new message form', () => {
    expect(component.find('NewMessageForm').length).toBe(1);
  });

  test('should match snapshot', () => {
    component = renderer.create(<Messages messages={messages} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
