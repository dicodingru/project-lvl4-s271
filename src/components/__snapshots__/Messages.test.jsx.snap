// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component should match snapshot 1`] = `
<div
  className="sc-ifAKCX bHMqJW"
>
  <div>
    <span>
      msg1
    </span>
  </div>
  <div>
    <span>
      msg2
    </span>
  </div>
  <div
    className="sc-bdVaJa UoSJW"
  >
    <form
      className="sc-bwzfXH uTxCW"
    >
      <input
        className="sc-htpNat jOwwja"
        type="text"
      />
      <button
        className="sc-bxivhb lbRlRl"
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
</div>
`;
